### PR TITLE
*: renaming variables/structs to improve readability / decrease confusion

### LINKED ISSUE #49988: PartitionInfo structs/variables are used for PartitionDefinitions and causing confusion when reading the code.
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

See [here](https://github.com/pingcap/tidb/blob/999f599d2fca4963ad97be65580d8e48a49e5a83/pkg/executor/point_get.go#L123), where:
```go
partInfo         *model.PartitionDefinition
```

### 2. What did you expect to see? (Required)
Either `partition`, `partitionDef` or something else, since `model.PartitionInfo` is a different struct, containing the full partition definitions.

### 3. What did you see instead (Required)
`partInfo` referring to a `PartitionDefinition`.

### 4. What is your TiDB version? (Required)

Current master, 999f599d2fca4963ad97be65580d8e48a49e5a83

### PR DESCRIPTION
**Pull request by mjonss**
### What problem does this PR solve?

Issue Number: close #49988

Problem Summary:
Naming of variables/structs causes confusion, since the names are not matching the type.

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [X] No need to test
  - Rename of variables only, no logic change.

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by YangKeao — APPROVED** [review]
LGTM

File: pkg/planner/core/point_get_plan.go [hunk 1]
```diff
@@ -315,7 +315,7 @@ type BatchPointGetPlan struct {
 	dbName           string
 	TblInfo          *model.TableInfo
 	IndexInfo        *model.IndexInfo
-	PartitionInfos   []*model.PartitionDefinition
+	PartitionDefs    []*model.PartitionDefinition
```
Comment (Defined2014) [inline pkg/planner/core/point_get_plan.go:318]: Why name it "Partition"? This is different from PointGet.

Comment (Defined2014) [reply pkg/planner/core/point_get_plan.go:318]: It's better to use `PartitionDefs` or `Partitions`.

Comment (mjonss) [reply pkg/planner/core/point_get_plan.go:318]: I now also changed `PointGetPlan`, to be `partitionDef`, so it is consistent.

**Review by Defined2014 — APPROVED** [review]

**Pull request merged**